plugins/cliccd.py

```diff
@@ -60,16 +61,22 @@ def _preprocessData(self, n, data, i):
         # positions scanned).
         return data.mean()
 
-    def _onCompletedData(self, n, raw_das):
+    def _assembleLiveData(self, n, raw_data, px_idx, rep, pol_idx):
+        if n != self._ccd_idx:
+            return super(SEMCLCCDStream, self)._assembleLiveData(n, raw_data, px_idx, rep, pol_idx)
+
+        raw_data.metadata[MD_DESCRIPTION] = self._streams[n].name.value
+        self._live_data[n].append(raw_data)
+
+    def _assembleFinalData(self, n, data):
         # Only override for the CCD data
         if n < len(self._streams) - 1:
-            r = super(SEMCLCCDStream, self)._onCompletedData(n, raw_das)
-            return r
+            return super(SEMCLCCDStream, self)._assembleFinalData(n, data)
 
         # Same as sem data, but without computing the data position from the
         # CCD metadata
         md = self._ccd_md.copy()
-        sem_data = self._raw[0]  # _onCompletedData() should be called in order
+        sem_data = self._raw[0]  # _assembleFinalData() should be called in order
         md[model.MD_POS] = sem_data.metadata[model.MD_POS]
         md[model.MD_DESCRIPTION] = self._streams[n].name.value
```

Comment:
You can drop this line, as you've put it in `_assembleLiveData()`.
I think that could also move the MD_POS and removal of the MD_AR* to `_assembleLiveData()`.